library/datasort.c

```diff
@@ -1668,6 +1668,7 @@ int eblob_generate_sorted_data(struct datasort_cfg *dcfg)
 
 	/* Lock backend */
 	pthread_mutex_lock(&dcfg->b->lock);
+	pthread_rwlock_wrlock(&dcfg->b->iteration_lock);
```

Comment:
Have you checked all branches and sure that there is no deadlock?

Reply:
Checked and didn't find suspicious places.

Reply:
Neither do I, thanks